File: MATTHEW/leetcode/AIcoding/3.DeepSeek-v3/go_solutions/2934-minimum-operations-to-maximize-last-elements-in-arrays.go
func minOperations(nums1 []int, nums2 []int) int {
    n := len(nums1)
    if n == 1 {
        return 0
    }

    count1, count2 := 0, 0
    last1, last2 := nums1[n-1], nums2[n-1]

    for i := 0; i < n-1; i++ {
        if nums1[i] <= last1 && nums2[i] <= last2 {
            continue
        } else if nums2[i] <= last1 && nums1[i] <= last2 {
            count1++
        } else {
            count1 = -1
            break
        }
    }

    for i := 0; i < n-1; i++ {
        if nums1[i] <= last2 && nums2[i] <= last1 {
            continue
        } else if nums2[i] <= last2 && nums1[i] <= last1 {
            count2++
        } else {
            count2 = -1
            break
        }
    }

    if count1 == -1 && count2 == -1 {
        return -1
    } else if count1 == -1 {
        return count2
    } else if count2 == -1 {
        return count1
    } else {
        return min(count1, count2)
    }
}

func min(a, b int) int {
    if a < b {
        return a
    }
    return b
}

File: leetcode/AIcoding/2.Gemini-2.0/erlang_solutions/2934-minimum-operations-to-maximize-last-elements-in-arrays.erl
-module(minimum_operations).
-export([min_operations/2]).

min_operations(Nums1, Nums2) ->
  N = length(Nums1),
  Last1 = lists:last(Nums1),
  Last2 = lists:last(Nums2),

  case check(Nums1, Nums2, N, Last1, Last2) of
    {inf, inf} -> -1;
    {A, B} -> min(A, B)
  end.

check(Nums1, Nums2, N, Last1, Last2) ->
  check_helper(Nums1, Nums2, N, Last1, Last2, 0, 0).

check_helper(Nums1, Nums2, N, Last1, Last2, Count1, Count2) ->
  case N of
    0 -> {Count1, Count2};
    _ ->
      H1 = hd(Nums1),
      H2 = hd(Nums2),
      T1 = tl(Nums1),
      T2 = tl(Nums2),

      case true of
        H1 =< Last1 and H2 =< Last2 ->
          check_helper(T1, T2, N - 1, Last1, Last2, Count1, Count2);
        H1 =< Last2 and H2 =< Last1 ->
          check_helper(T1, T2, N - 1, Last1, Last2, Count1 + 1, Count2 + 1);
        H1 =< Last1 and H2 > Last2 ->
          check_helper(T1, T2, N - 1, Last1, Last2, inf, Count2 + 1);
        H1 > Last1 and H2 =< Last2 ->
          check_helper(T1, T2, N - 1, Last1, Last2, Count1 + 1, inf);
        true ->
          {inf, inf}
      end
  end.